homeassistant/components/sensibo/entity.py

```diff
@@ -0,0 +1,75 @@
+"""Base entity for Sensibo integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import async_timeout
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, LOGGER, SENSIBO_ERRORS, TIMEOUT
+from .coordinator import SensiboDataUpdateCoordinator
+
+
+class SensiboBaseEntity(CoordinatorEntity):
+    """Representation of a Sensibo numbers."""
+
+    coordinator: SensiboDataUpdateCoordinator
+
+    def __init__(
+        self,
+        coordinator: SensiboDataUpdateCoordinator,
+        device_id: str,
+    ) -> None:
+        """Initiate Sensibo Number."""
+        super().__init__(coordinator)
+        self._device_id = device_id
+        self._client = coordinator.client
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, coordinator.data[device_id]["id"])},
+            name=coordinator.data[device_id]["name"],
+            connections={(CONNECTION_NETWORK_MAC, coordinator.data[device_id]["mac"])},
+            manufacturer="Sensibo",
+            configuration_url="https://home.sensibo.com/",
+            model=coordinator.data[device_id]["model"],
+            sw_version=coordinator.data[device_id]["fw_ver"],
+            hw_version=coordinator.data[device_id]["fw_type"],
+            suggested_area=coordinator.data[device_id]["name"],
```

Comment:
Each one of these still has to do a lookup so its a bit more efficient to do it once above